src/Yaapii.Xambly/Arg/NotIllegalText.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Yaapii.Atoms;
+using Yaapii.Atoms.Scalar;
+
+namespace Yaapii.Xambly.Arg
+{
+    /// <summary>
+    /// A text with only legal xml characters
+    /// </summary>
+    public sealed class NotIllegalText : IText
+    {
+        private readonly IScalar<string> text;
+
+        /// <summary>
+        /// A text with only legal xml characters
+        /// </summary>
+        public NotIllegalText(string text)
+        {
+            this.text = new ScalarOf<string>(() =>
```

Comment:
this should be sticky. The injected text can never change.

Reply:
ScalarOf is sticky in the new Atoms